Content.Shared/Storage/EntitySystems/MagnetPickupSystem.cs

```diff
@@ -48,17 +50,21 @@ public override void Update(float frameTime)
 
             comp.NextScan += ScanDelay;
 
-            if (!_inventory.TryGetContainingSlot((uid, xform, meta), out var slotDef))
-                continue;
+            var parentUid = xform.ParentUid;
 
-            if ((slotDef.SlotFlags & comp.SlotFlags) == 0x0)
-                continue;
+            if (comp.SlotFlags != null || !_hands.TryGetActiveItem(parentUid, out var activeItem) && activeItem != uid)
```

Comment:
This should be its own bool for more control. You're right side logic here also always evaluates false since `!_hands.TryGetActiveItem()` will be false when it finds an item, but `activeItem != uid` will be false if there's nothing in hand.

```suggestion
            if (comp.RequireActiveHand && (!_hands.TryGetActiveItem(parentUid, out var activeItem) || activeItem != uid))
                continue;
                
            if (comp.SlotFlags != null)
```